### PR TITLE
Learning Goals: Implement Entities and Database Changes

### PR DESCRIPTION
**Pull request by stefanwaldhauser**
### Checklist
- [X] Server: I documented the Java code using JavaDoc style.

### Motivation and Context

One part of my bachelor thesis is the implementation of the concept of learning goals. An instructor can define learning goals and connect them to lectures and exercises. Later this will be used to calculate a student's progress in learning goals. 

Graphical Representation (ignore lecture content part for this PR)
image

### Description

Implemented necessary entities and their relationships for learning goals

### Steps for Testing

- Help me by looking at the code and tell me if you notice anything that is wrong concerning JPA best practices

### Test Coverage
- No testing yet possible as this contains just the entities

### CODE REVIEW

**Review by MaisiKoleni — COMMENTED** [review]
Code looks good to me.

Is the `LearningGoalRepository` reserved for a future PR? I think a very basic implementation of that would fit well in here, too.

**Comment by stefanwaldhauser** [comment]
> Code looks good to me.
> 
> Is the `LearningGoalRepository` reserved for a future PR? I think a very basic implementation of that would fit well in here, too.

I am splitting it up into small PRs that make it easy to review. As I am not sure what querys i really need i decided not to implement any in this PR yet

**Review by MaisiKoleni — APPROVED** [review]
> > Code looks good to me.
> > Is the `LearningGoalRepository` reserved for a future PR? I think a very basic implementation of that would fit well in here, too.
> 
> I am splitting it up into small PRs that make it easy to review. As I am not sure what querys i really need i decided not to implement any in this PR yet

I just thought of an "empty" repository interface so you can - in theory -  save, get and delete.

But more, smaller PR are fine as well for me.

File: src/main/java/de/tum/in/www1/artemis/domain/Exercise.java [hunk 1]
```diff
@@ -523,7 +524,7 @@ else if (participation.getExercise() instanceof ModelingExercise || participatio
      * Get the latest relevant result from the given participation (rated == true or rated == null) (relevancy depends on Exercise type => this should be overridden by subclasses
      * if necessary)
      *
-     * @param participation the participation whose results we are considering
+     * @param participation           the participation whose results we are considering
```
Comment (FrankeLukas) [inline src/main/java/de/tum/in/www1/artemis/domain/Exercise.java:527]: This whitespace change doesn't look correct to me.

**Review by FrankeLukas — APPROVED** [review]
I'm not too familiar with this code, but the changes look logical to me.

**Comment by stefanwaldhauser** [comment]
Will be continued in: https://github.com/ls1intum/Artemis/pull/2178

**Pull request closed without merging**